pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4237,6 +4237,17 @@ protected void internalGetLastMessageId(AsyncResponse asyncResponse, boolean aut
             });
     }
 
+    protected CompletableFuture<Void> internalRemoveSubscriptionTypesEnabled() {
+        return getTopicPoliciesAsyncWithRetry(topicName)
+                .thenCompose(op -> {
+                    if (!op.isPresent()) {
+                        return CompletableFuture.completedFuture(null);
+                    }
+                    op.get().setSubscriptionTypesEnabled(Lists.newArrayList());
```

Comment:
We should not modify a object that could be the result of a read from a cache

Reply:
Sorry, I don't fully understand what you mean. I'm still a little confused about it. Can you give me more information？
I understand that this is the process of making `TopicPolicies.subscriptionTypesEnabled` as `empty list`(default initial value), right?
Maybe according to your meaning, how should I modify it?

Reply:
you are using the returned object from `getTopicPoliciesAsyncWithRetry` using `op.get()`
if `pulsar().getTopicPoliciesService().updateTopicPoliciesAsync` fails then the local copy of the variable in memory is update but the value is not write to persistent metadata storage.

We should copy/clone the object and not mutate it.
It would be better to have "immutable" objects, but that would be another story

Reply:
Many topic policy `set`&`remove` operations are implemented in this way, so they all need to be adjusted, as below:
https://github.com/apache/pulsar/blob/aa97d39e14ba0b26ce8975bf9490c709931753c5/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java#L3012-L3020
https://github.com/apache/pulsar/blob/aa97d39e14ba0b26ce8975bf9490c709931753c5/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java#L3022-L3031